packages/opentelemetry-types/src/metrics/gauge.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface GaugeTimeseries {
+  // Adds the given value to the current value. Values can be negative.
+  add(value: number): void;
```

Comment:
What is the purpose of `add`? A gauge by definition can only be set. If incrementing and decrementing is wanted, then a counter should probably used instead.

Reply:
I'm not entirely sure, it looks to be pretty close to what census had.

Reply:
OpenCensus has `inc` and `reset` for counters and `add` and `inc` for gauges.

Reply:
Ah cool was looking at [this](https://github.com/census-instrumentation/opencensus-node/blob/master/packages/opencensus-core/src/metrics/gauges/gauge.ts#L189), didn't see the impl for cumulative. 

There's currently an RFC from @jmacd to change the verbiage for metrics, along with removing meter.record https://github.com/open-telemetry/rfcs/pull/4/files#diff-a4b53eafb1f61334abc351feb3ecc5b3R20

Reply:
This is the impl for cumulative: https://github.com/census-instrumentation/opencensus-node/blob/master/packages/opencensus-core/src/metrics/cumulative/cumulative.ts#L196

Reply:
I would agree that `add` feels a bit weird for gauges, but I could also imagine cases where you want it. I gave the example above of some resource budget that you can both deposit and withdraw from (and maybe even go negative with). That metric type I would consider a gauge because we allow it to be negative and it's more like measuring the amount of something available at a given time (like temperature) instead of counting occurrences of an event stream (like request count)